docs/deployment-guide.md

```diff
@@ -213,20 +213,82 @@ If you have a valid license place your `.lic` file in the `configuration_files/l
 
-### Alfresco/Solr authentication
+### Secrets management
 
-As of ACS 7.2 and/or Search services 2.0.3, the repository <--> solr communication requires to be authenticated. The playbook will set up that authentication scheme using the new `secret` method.
-This methods needs to be passed a shared secret. In order to do so use the variable below:
+This playbook expects that security-relevant variables are configured in `vars/secrets.yml` file.
 
-```yaml
-reposearch_shared_secret: dummy
+Is it strongly recommended to enable [Ansible Vault](https://docs.ansible.com/ansible/latest/user_guide/vault.html) in order to avoid keeping secrets in plaintext on the control node file-system.
+
+If you run the playbook without configuring Ansible Vault, it will automatically
+generate those secrets but they will be saved as plaintext.
+
+#### Enable Ansible Vault support
+
+The steps required to enable **Ansible Vault** support are:
+
+* Configure a vault password in a file (e.g. `~/.vault_pass.txt`)
+  * Optionally you can autogenerate a strong password with:
+
+    ```bash
+    openssl rand -base64 21 > ~/.vault_pass.txt
+    ```
+
+* Export the path to the vault password file as `ANSIBLE_VAULT_PASSWORD_FILE` so
+  that can automatically picked-up when running Ansile:
+
+  ```bash
+  export ANSIBLE_VAULT_PASSWORD_FILE=~/.vault_pass.txt
+  ```
+
+Otherwise you can provide the password before each playbook invocation with:
+
+```bash
+ansible-playbook --ask-vault-pass ...
+```
+
+Now you are ready to start using Ansible Vault.
+
+#### Populate secrets with Ansible Vault
+
+Ansible Vault provides two alternative ways to protect secrets:
+
+* [Encrypted variables](https://docs.ansible.com/ansible/latest/user_guide/vault.html#encrypting-individual-variables-with-ansible-vault)
+* [Encrypted files](https://docs.ansible.com/ansible/latest/user_guide/vault.html#encrypting-files-with-ansible-vault)
+
+In the previous links you can read both advantages and disadvantages of the two approaches.
+
+We provide a script to automatically autogenerate secure secrets (`openssl` is required).
+
+If you prefer **Encrypted variables** run:
+
+```bash
+./scripts/generate-secrets.sh > vars/secrets.yml
 ```
 
-> Of course do not use dummy as shown above, but use a stronger secret
+or if you prefer **Encrypted files** run:
+
+```bash
+./scripts/generate-secrets.sh plaintext > vars/secrets.yml
+ansible-vault encrypt vars/secrets.yml
+```
+
+Now mandatory secrets are ready to use by the playbook.
+
+Please refer to the [official documentation](https://docs.ansible.com/ansible/latest/user_guide/vault.html) to learn how to interact with existing encrypted variables or files.
+
+#### Populate secrets with third-party solutions
+
+Variables defined in `vars/secrets.yml` can also be populated dynamically using third-parties lookup plugins:
```

Comment:
we can review this together, from my understanding when the integration is via lookup plugins, the user will just declare in vars/secrets.yml something like:

```yml
activemq_password: lookup('1password', 'my/activemq/password')
```

so for the current state of the playbook, it's transparent: playbook will not try to add ansible-vault credentials because the keys are already defined in secrets.yml.

however if we expect that the user should hook a custom playbook before ours, that maybe set credentials as facts, we probably need a kill switch for out entire ansible-vault integration?

Reply:
let's discuss that yes, but tbh I forgot those were lookup plugin, so that might actually be ok